paimon-common/src/main/java/org/apache/paimon/utils/BinaryStringUtils.java

```diff
@@ -315,6 +322,25 @@ public static Timestamp toTimestamp(BinaryString input, int precision, TimeZone
         return DateTimeUtils.parseTimestampData(input.toString(), precision, timeZone);
     }
 
+    // Helper method to convert milliseconds to Timestamp with the provided precision.
+    private static Timestamp fromMillisToTimestamp(long millis, int precision) {
+        // Calculate seconds and nanoseconds from the millis
+        long seconds = millis / 1000;
+        int nanos = (int) ((millis % 1000) * 1_000_000);
+
+        // Adjust the nanoseconds to the specified precision
+        if (precision < 9) {
+            nanos =
+                    (int)
+                            (Math.floor(nanos / Math.pow(10, 9 - precision))
+                                    * Math.pow(10, 9 - precision));
+        }
```

Comment:
I think this is unnecessary. We can store the data and let the readers truncate the data based on precision.
So we can just use `Timestamp#fromEpochMillis(long milliseconds, int nanosOfMillisecond)`

Reply:
I read the documentation for `Timestamp#fromEpochMillis(long milliseconds, int nanosOfMillisecond)` and it expects nanosOfMs to be in range [0, 999_999]. Are you saying we need to default to precision 6 here?

Reply:
For example, if we want to store `1970-01-01 00:00:00.123456789`, the first argument `milliseconds` is `123` and the second argument `nanosOfMillisecond` is `456789`.

Reply:
@yuzelin done.

Reply:
Thanks~